### PR TITLE
Modify MessagingThreadListItem to optionally display an alert icon

### PR DESCRIPTION
**Pull request by dengjulie**
**Jira:**
https://clever.atlassian.net/browse/PRTL-241

**Overview:**
Teachers Inviting Guardians aka teavites project:
We want to display an alert icon in the thread list for contacts whose invites have expired or been rejected, so that the teacher knows to take action.
image

**Screenshots/GIFs:**
Screen Shot 2021-02-28 at 5 25 56 PM

**Testing:**

- [ ] Unit tests
- Manual tests:
  - [ ] Chrome
  - [ ] Safari
  - [ ] IE11

**Roll Out:**

- Before merging:
  - [x] Updated docs
  - [x] Bumped version in `package.json`
    - Breaking change?
      - If it is a beta component run `npm version minor`
      - If the component is not in beta run `npm version major`
    - New component or backward-compatible component feature change? Run `npm version minor`
    - Only changing documentation? All good. Skip this step.
  - After creating a new component, make sure to add it to the Components List in `ComponentsView.jsx`. To do so:
    - [ ] Add an entry in `ComponentsView.componentsToDisplay` using this template:
      ```
      {
        componentLink: "<COMPONENT LINK>",
        componentImg: "<COMPONENT LINK>.png",
        componentName: "<COMPONENT NAME>",
        componentImgAlt: "A <COMPONENT NAME> component",
      },
      ```
    - [ ] Add a screenshot of the component in `docs/assets/img` with the format `<COMPONENT LINK>.png`
- After merging:
  - [ ] Deployed updated docs (`make deploy-docs`)
  - [ ] Posted in #eng if I made a breaking change to a beta component

### CODE REVIEW

File: src/MessagingThreadListItem/MessagingThreadListItem.tsx [hunk 1]
```diff
@@ -110,10 +119,11 @@ export const MessagingThreadListItem: React.FC<
           {subContent && (
             <FlexBox alignItems={ItemAlign.CENTER}>
               <div className={subContentClass}>{subContent}</div>
-              {indicator}
+              {timestamp && indicator}
```
Comment (TheRealBenjiLi) [inline src/MessagingThreadListItem/MessagingThreadListItem.tsx:122]: Why do we only want to show the indicator when a timestamp is around? :O

File: src/MessagingThreadListItem/MessagingThreadListItem.tsx [hunk 2]
```diff
@@ -110,10 +119,11 @@ export const MessagingThreadListItem: React.FC<
...
             </FlexBox>
           )}
         </FlexItem>
+        <FlexItem alignItems={ItemAlign.CENTER}>{subContent && !timestamp && indicator}</FlexItem>
```
Comment (TheRealBenjiLi) [inline src/MessagingThreadListItem/MessagingThreadListItem.tsx:126]: If I'm reading this right, this is rendering the indicator. But isn't that also being done on line 122?

File: src/MessagingThreadListItem/MessagingThreadListItem.tsx (hunk 1 above)
Comment (dengjulie) [reply src/MessagingThreadListItem/MessagingThreadListItem.tsx:122]: So we have 3 options: either display timestamp + indicator, just timestamp, or just indicator. There is no instance of messaging using just an indicator yet, so this is a new use case for invites (mocks don't have timestamp, and we don't sort threads by order of time invites are sent out). With just timestamp, it looks fine to have it placed in the same horizontal axis of the title, but with just indicator, it looks a bit funky for it to be on the lower right of the thread list item. we want it vertically centered as another column of the overall parent flexbox on the right side of the thread item. Does that make sense?

File: src/MessagingThreadListItem/MessagingThreadListItem.tsx (hunk 2 above)
Comment (dengjulie) [reply src/MessagingThreadListItem/MessagingThreadListItem.tsx:126]: Commented above, but maybe this is too confusing at first glance... I'll think of how to make it more clear.

**Comment by dengjulie** [comment]
Pause on this, going to ask Jemma some design Qs

File: src/MessagingThreadListItem/MessagingThreadListItem.tsx (hunk 1 above)
Comment (TheRealBenjiLi) [reply src/MessagingThreadListItem/MessagingThreadListItem.tsx]: Ah yep, that clears things up! :)

**Review by TheRealBenjiLi — APPROVED** [review]
🌮

**Pull request merged**